File: library/src/main/java/com/licrafter/tagview/utils/DipConvertUtils.java
package com.licrafter.tagview.utils;

import android.content.Context;

/**
 * author: shell
 * date 2016/12/22 下午1:04
 **/
public class DipConvertUtils {

    public static int dip2px(Context context, float dpValue) {
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }
}

File: library/src/main/kotlin/com/licrafter/tagview/utils/DipConvertUtils.kt
package com.licrafter.tagview.utils

import android.content.Context

/**
 * Created by lijx on 2018/5/12.
 * Gmail: shellljx@gmail.com
 */
object DipConvertUtils {

    @JvmStatic
    fun dip2px(context: Context, dpValue: Float): Int {
        val scale = context.resources.displayMetrics.density
        return (dpValue * scale + 0.5f).toInt()
    }
}